### PR TITLE
fix(parental-leave): use dob instead of estimated dob sometimes

### PR DESCRIPTION
**Pull request by karenbjorg**
[TS-392](https://dit-iceland.atlassian.net/jira/software/c/projects/TS/boards/82?assignee=62554189c23e5b006ab32a3d&selectedIssue=TS-392)

## What

If baby has been born then sometimes use the date of birth instead of estimated date of birth.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Added support for handling birth dates in parental leave date calculations.

- **Enhancements**
  - Improved date calculation logic to include birth dates in addition to expected dates of birth or adoption.

- **Localization**
  - Added new message key for "Date of birth" in the parental leave form messages.

### CODE REVIEW

**Review by ylfahfa — APPROVED** [review]
lgtm 🤠

File: libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx [hunk 1]
```diff
@@ -104,6 +109,9 @@ const InReviewSteps: FC<React.PropsWithChildren<FieldBaseProps>> = (props) => {
               ? formatMessage(
                   parentalLeaveFormMessages.reviewScreen.adoptionDate,
                 )
+              : periods?.[0]?.firstPeriodStart ==
```
Comment (veronikasif) [inline libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx:112]:
```suggestion
              : periods?.[0]?.firstPeriodStart ===
```
Expected '===' and instead saw '=='. eslint

File: libs/application/templates/parental-leave/src/lib/messages.ts [hunk 2]
```diff
@@ -1818,6 +1818,11 @@ export const parentalLeaveFormMessages = {
       defaultMessage: 'Forsjárskipting á sér stað',
       description: 'Division of custody takes place',
     },
+    birthDate: {
+      id: 'pl.application:review.birth.date',
+      defaultMessage: 'Fæðingardagur',
+      description: 'Date of birth',
+    },
```
Comment (veronikasif) [inline libs/application/templates/parental-leave/src/lib/messages.ts:1825]: This text exists in `shared.dateOfBirthTitle`

**Review by veronikasif — APPROVED** [review]
LGMT 👍

**Pull request merged**